File: src/ca/rightsomegoodgames/mayacharm/actions/BaseSendAction.java
package ca.rightsomegoodgames.mayacharm.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.LangDataKeys;
import com.intellij.openapi.fileTypes.FileTypeManager;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public abstract class BaseSendAction extends AnAction {
    private final FileTypeManager fileTypeManager;

    public BaseSendAction() {
        fileTypeManager = FileTypeManager.getInstance();
    }

    public abstract void actionPerformed(@NotNull AnActionEvent e);

    @Override
    public void update(AnActionEvent e) {
        super.update(e);
        VirtualFile currentFile = e.getData(LangDataKeys.VIRTUAL_FILE);
        if (currentFile == null) {
            e.getPresentation().setEnabled(false);
            return;
        }

        if (fileTypeManager.isFileOfType(currentFile, fileTypeManager.getFileTypeByExtension(".py")))
            e.getPresentation().setEnabled(true);
        else
            e.getPresentation().setEnabled(false);
    }
}

File: src/ca/rightsomegoodgames/mayacharm/actions/BaseSendAction.kt
package ca.rightsomegoodgames.mayacharm.actions

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.LangDataKeys
import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.fileTypes.FileTypeManager
import com.intellij.openapi.project.DumbAwareAction

abstract class BaseSendAction : DumbAwareAction() {
    companion object {
        private val fileTypeManager: FileTypeManager = FileTypeManager.getInstance()
        private val pyFileType: FileType = fileTypeManager.findFileTypeByName("Python")!!
    }

    abstract override fun actionPerformed(e: AnActionEvent)

    override fun update(e: AnActionEvent) {
        val currentFile = e.getData(LangDataKeys.VIRTUAL_FILE)
        if (currentFile == null) {
            e.presentation.isEnabled = false
            return
        }

        e.presentation.isEnabled = fileTypeManager.isFileOfType(currentFile, pyFileType)
    }
}
